packages/mui-material/src/TablePagination/TablePagination.js

```diff
@@ -297,10 +347,6 @@ TablePagination.propTypes /* remove-proptypes */ = {
    * Override or extend the styles applied to the component.
    */
   classes: PropTypes.object,
-  /**
-   * @ignore
-   */
-  className: PropTypes.string,
```

Comment:
Is this expected?

Reply:
Yes, this is generated from the proptypes script.

Reply:
Right, I understand that. My question was, why in this PR related to adding slots, is the `className` being removed from the proptypes?

I see it was probably due to `className` being removed from the props destructuring [here](https://github.com/mui/material-ui/pull/44570/files#diff-48a78bf0f365533c6760c13adf35c38d3b1cd3d3c0476f31d60785f0e11d8492L155)?

If so, that's ok. My only final question would be, if an user was providing `className` to this component, would this change make it throw a console warning?

Reply:
> If so, that's ok. My only final question would be, if an user was providing className to this component, would this change make it throw a console warning?

No, it won't throw a warning. Tested with this [sandbox](https://codesandbox.io/p/sandbox/material-ui-cra-ts-forked-rsy7s4).